### PR TITLE
Feature developer stats

### PR DESCRIPTION
**Pull request by nsimaria**
Added `./bin/developer-stats.js` command to publish developer and repo stats to Slack.

### CODE REVIEW

File: lib/CheckFourEyesCommander.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            fs.writeFile(filename, content, function(err) {
+                if(err) {
+                    throw err;
+                }
+
+                console.log (okCounter + ' pull requests are merged by someone other than the author.');
+                console.log (koCounter + ' pull requests are merged by the author.');
+                console.log('report saved on ' + filename);
+
+                return true;
```
Comment (PepijnSenders) [inline lib/CheckFourEyesCommander.js]: No need for this.

File: lib/DeveloperStatsCommander.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                }
+
+                if (repo.name.indexOf('-test') > -1 && repo.name.indexOf('-tests') == -1)
+                    return false;
+
+                return true;
+            });
+
+            return Q.allSettled (repos.map (function (repo) {
+//                console.log (repo); return;
```
Comment (PepijnSenders) [inline lib/DeveloperStatsCommander.js]: http://butunclebob.com/ArticleS.TimOttinger.ApologizeIncode

File: lib/DeveloperStatsCommander.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+            this.slack (repoStats);
+            this.slack (contributorStats);
+        }
+
+        return;
+    }
+
+    DeveloperStatsCommander.prototype.slack = function (message) {
+
+        var sh = 'curl -X POST --data-urlencode \'payload={"username": "' + process.env.SLACK_USERNAME + '", "text": "'+ message + '", "icon_emoji": "' + process.env.SLACK_EMOJI + '"}\' ' + process.env.SLACK_URL;
```
Comment (PepijnSenders) [inline lib/DeveloperStatsCommander.js]: You can use http://sugarjs.com/api/String/assign, to make this a bit more readable.

File: lib/DeveloperStatsCommander.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            }
+
+            if (totalCommits == 0) {
+                zeroes.push (p);
+                continue;
+            }
+
+            notes = '(' + notes + ')';
+
+            line += '`' + ('   ' + totalCommits).substr(-3) + '` ';
```
Comment (PepijnSenders) [inline lib/DeveloperStatsCommander.js]: Maybe we should use some kind of templating engine for this like `mustache`. Inline template is bad, even in JavaScript :wink:

File: lib/DeveloperStatsCommander.js (hunk 3 above)
Comment (nsimaria) [inline lib/DeveloperStatsCommander.js]: i'll do that on a later iteration

File: lib/DeveloperStatsCommander.js (hunk 4 above)
Comment (nsimaria) [inline lib/DeveloperStatsCommander.js]: i'll do that on a later iteration

**Pull request merged**